test/redis_storage_case.exs

```diff
@@ -0,0 +1,289 @@
+defmodule CxLeaderboard.RedisStorageCase do
```

Comment:
Is there a reason why RedisStorageCase is needed as opposed to reusing the existing StorageCase?

Reply:
yeah, so this is the tip of the iceberg with this PR, and I'm curious to get your thoughts. The Redis implementation necessitated some differences in semantics since sorting does not happen as a separate operation (it happens atomically as things are added to the sorted set). There are also some other important changes that aren't top of mind for me, but would be happy to point out once i take a closer look again.

Reply:
To answer your question though, it was a clean way of pursuing this branch without breaking existing functionality. I think you're right that in the future this could go away potentially.

Reply:
Ah, I see. I'm happy to re-evaluate what we're testing for. For example, I notice you left a comment that it doesn't make sense to get a range in reverse order.

My goal here is to try and make the public interface more or less consistent, and purely Leaderboard focused, regardless of storage engine. I would rather we remove some of the not-very-useful functionality (like you did with the reverse range test) than have different storage engines provide different behavior.

Would you mind going over each alteration you had to make between this and default StorageCase, and providing some reasons why each was necessary? In the end it might make sense to clean up StorageCase to accommodate RedisStore better.